src/bootstrap/step.rs

```diff
@@ -814,6 +814,9 @@ invalid rule dependency graph detected, was a rule added and maybe typo'd?
             Subcommand::Clean => panic!(),
         };
 
+        // Only construct our plan from build triple steps for dist builds.
+        let from_build_only = kind == Kind::Dist;
```

Comment:
Could this actually get moved down to the `let hosts` statement? It can have a comment like:

```rust
// For 'dist' steps we only distribute artifacts built from
// the build platform, so only consider that in the hosts
// array.
```